src/Microsoft.ML.Vision/ImageClassificationTrainer.cs

```diff
@@ -630,8 +638,9 @@ private protected override ImageClassificationModelParameters TrainModelCore(Tra
                     _inputTensorName, _bottleneckTensor.name, trainSetBottleneckCachedValuesFilePath,
                     ImageClassificationMetrics.Dataset.Train, _options.MetricsCallback);
 
+                string sizeFile = Path.Combine(_options.WorkspacePath, "TrainingSetSize.txt");
```

Comment:
(minor) you may want to extract this into a helper method so it isn't duplicated where this file is, and what its name is.